### PR TITLE
House Price Prediction using Machine Learning

### PR DESCRIPTION
**Pull request by lalith1403**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by lalith1403** [comment]
This fixes #474

**Comment by hectorkambow** [comment]
Plagiarism check done

**Comment by adrianmurage** [comment]
Hi @lalith1403
I'll be conducting the review in two stages:
- Preliminary review where I'll look out for: grammar, spelling, and readability
- Tech Review where I'll look out for: technical accuracy, beginner-friendliness, and currency

File: articles/house-price-prediction/index.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+We need to convert this into a numerical value. To do this, we can use the `LabelEncoder` function available in the pre-processing module of sklearn. This will replace the column with numerical values of 0 and 1, respectively. 0 indicates `Near,` and 1 indicated `Far.`
+
+Once we perform the pre-processing of the dataset, we can fit the data to the model. We begin with instantiating an object of the `RandomForestRegressor` class. This is available in the `sklearn.ensemble` module. We use the `fit` method to fit the data to the model.
+
+Once the model is fit, we evaluate the model’s performance on the test set we got earlier. We use the `predict` method present in the `RandomForestRegressor` class. The `predict` method takes in the test input data and predicts an output. Using the predicted output and the actual output known from the dataset, we compute the test accuracy.
+
+Another useful evaluation metric is the `Mean Squared Error`. The MSE loss gives an estimate of how far the prediction is from the mean of the output. Computing the MSE gives us an idea about the performance of the algorithm.
```
Comment (adrianmurage) [inline articles/house-price-prediction/index.md:71]: Consider adding the abbreviation next to the text to explicitly relate them:
`Mean Squared Error`(MSE)

File: articles/house-price-prediction/index.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+**lstat** | *lower status of the population (percent).*
+**medv** | *median value of owner-occupied homes in \$1000s.*
+
+
+### Approach taken
+We'll use the Random Forest regression algorithm to predict the price of the houses. In this article, we'll consider machine learning algorithms as a black box that fits the data. 
+
+This article focuses more on the machine learning pipeline. For more information on the Random Forest algorithm, I suggest looking into this [video](https://www.youtube.com/watch?v=nxFG5xdpDto).
+
+We'll begin with loading the data. Since we are using an inbuilt dataset, we'll be calling the `load_boston` function from the `sklearn.datasets` module. The data is loaded into the `data` variable. Once the data is loaded, we separate the data and target attributes of the `data` variable. We store them in variables `data` and `target` respectively.
```
Comment (adrianmurage) [inline articles/house-price-prediction/index.md:55]: Consider changing the preposition "with" It should be:
"We'll begin by loading the data."

File: articles/house-price-prediction/index.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+
+# print the final results
+print("Training Accuracy:",dt.score(Xtrain,Ytrain))
+print("Testing Accuracy:",accuracy)
+print("Mean Squared Error",MSE_score.mean())
+```
+
+The code output should look something like this:
+
+![output](/engineering-education/house-price-prediction/output.jpg)
```
Comment (adrianmurage) [inline articles/house-price-prediction/index.md:133]: You could consider copy-pasting the terminal output and placing it in a code fence with bash as the language.

**Review by adrianmurage — CHANGES_REQUESTED** [review]
The article reads well. Great job @lalith1403.
Make the minor changes I have highlighted as I move on to the tech review.

File: articles/house-price-prediction/index.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  - url: /engineering-education/house-price-prediction/hero.jpg
+    alt: house price prediction example image
+---
+Welcome to a tutorial on predicting house prices using the [Random Forest Regression](https://en.wikipedia.org/wiki/Random_forest) algorithm. We will cover the data pipeline creation. This pipeline creation process involves loading the dataset, cleaning and pre-processing the dataset, fitting a model to the dataset, and testing the model’s performance using various evaluation metrics.
+<!--more-->
+### Prerequisites
+There are a few theoretical and programming pre-requisites required for this article. They are helpful but not required to understand the article.
+
+1. [Introduction to Pandas](/engineering-education/data-analytics-using-pandas/)
+2. [Introduction to Supervised Learning Algorithms using Scikit-Learn](/engineering-education/supervised-learning-algorithms/)
```
[the remark this answers is not in the document]
Comment (lalith1403) [reply articles/house-price-prediction/index.md:23]: This will work when published. It is a relative link.

[the remark this answers is not in the document]
Comment (adrianmurage) [reply articles/house-price-prediction/index.md:23]: Right... like images.
Thanks for pointing that out

File: articles/house-price-prediction/index.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+Another useful evaluation metric is the `Mean Squared Error`. The MSE loss gives an estimate of how far the prediction is from the mean of the output. Computing the MSE gives us an idea about the performance of the algorithm.
+
+### Code
+We'll use the dataset that is stored in a CSV file for a detailed view and easier access. Download the CSV file from this [link](https://drive.google.com/file/d/1clV931HTopTlD7ZWLotFSbsr9SAX50S8/view?usp=sharing). Save the CSV file in the same directory as the python file.
+
+```py
+# load the libraries
+import sklearn
+from sklearn.metrics import mean_squared_error as MSE
+from sklearn. pre-processing import LabelEncoder
```
Comment (adrianmurage) [inline articles/house-price-prediction/index.md:80]: syntax error

**Review by adrianmurage — CHANGES_REQUESTED** [review]
@lalith1403 
I ran into a syntax error that I have highlighted here
But once fixed the code runs well

Once you make the changes, ping me
I'll do one last read and label the article ready for final review

**Review by adrianmurage — APPROVED** [review]

**Comment by adrianmurage** [comment]
@lalith1403 Great job!
@hectorkambow this article is ready for final review
